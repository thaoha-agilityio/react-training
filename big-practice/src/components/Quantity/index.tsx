import { memo } from 'react';
import { Button, Flex, Text } from '@chakra-ui/react';

type QuantityProps = {
  count?: number;
  onIncreaseProduct?: () => void;
  onDecreaseProduct?: () => void;
};

const Quantity = ({ count = 1, onIncreaseProduct, onDecreaseProduct }: QuantityProps) => (
  <Flex
    flexDir='row'
    alignItems='center'
    borderColor='gray.100'
    justifyContent='space-between'
    rounded='md'
    border='1px'
    w={{ base: '80px', md: '123px' }}
    h={{ base: '40px', md: '64px' }}
  >
    <Button variant='unstyled' onClick={onIncreaseProduct}>
      +
    </Button>
    <Text fontSize={{ base: 'xs', md: 'sm' }}>{count}</Text>
    <Button variant='unstyled' onClick={onDecreaseProduct}>
      -
    </Button>
  </Flex>
);

export default memo(Quantity);
