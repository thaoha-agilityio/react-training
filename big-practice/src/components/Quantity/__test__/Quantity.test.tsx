import { render, fireEvent, screen } from '@testing-library/react';

import Quantity from '../index';

const mockProps = {
  onIncreaseProduct: jest.fn(),
  onDecreaseProduct: jest.fn(),
};

describe('Quantity component', () => {
  test('Quantity component displays the default count correctly', () => {
    render(<Quantity {...mockProps} />);
    const countText = screen.getByText('1'); // Default count value is 1

    expect(countText).toBeInTheDocument();
  });

  test('Quantity component displays the provided count correctly', () => {
    render(<Quantity count={5} {...mockProps} />);
    const countText = screen.getByText('5'); // Provided count value is 5

    expect(countText).toBeInTheDocument();
  });

  test('Quantity component calls onIncreaseProduct when the "+" button is clicked', () => {
    render(<Quantity count={1} {...mockProps} />);
    const increaseButton = screen.getByText('+');

    fireEvent.click(increaseButton);

    expect(mockProps.onIncreaseProduct).toHaveBeenCalledTimes(1);
  });

  test('Quantity component calls onDecreaseProduct when the "-" button is clicked', () => {
    render(<Quantity count={1} {...mockProps} />);
    const decreaseButton = screen.getByText('-');

    fireEvent.click(decreaseButton);

    expect(mockProps.onDecreaseProduct).toHaveBeenCalledTimes(1);
  });
});
