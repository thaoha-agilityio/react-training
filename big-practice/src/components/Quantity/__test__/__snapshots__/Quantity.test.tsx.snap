// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity component Quantity component displays the default count correctly 1`] = `
<div>
  <div
    class="css-ddi599"
  >
    <button
      class="chakra-button css-4xx2wk"
      type="button"
    >
      -
    </button>
    <p
      class="chakra-text css-1il0tvo"
    >
      1
    </p>
    <button
      class="chakra-button css-4xx2wk"
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
